client/lxd_network_allocations.go

```diff
@@ -0,0 +1,27 @@
+package lxd
+
+import (
+	"github.com/canonical/lxd/shared/api"
+)
+
+// GetNetworkAllocations returns a list of Network allocations tied to one or several projects (e.g, for IPAM information for example).
+func (r *ProtocolLXD) GetNetworkAllocations(allProjects bool) ([]api.NetworkAllocations, error) {
+	err := r.CheckExtension("network_allocations")
+	if err != nil {
+		return nil, err
+	}
+
+	netAllocations := []api.NetworkAllocations{}
+	uri := "/network-allocations"
+	if allProjects {
+		uri += "?all=true"
```

Comment:
Existing API endpoints that support all projects use the `all-projects` query parameter. We should use that here for consistency and clarity.